src/controllers/wallet/tx-proposal/tx-proposal.controller.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const {
+  constants: hathorLibConstants,
+  helpersUtils,
+  SendTransaction,
+  transaction,
+  transaction: transactionUtils,
+  walletUtils,
+  storage,
+} = require('@hathor/wallet-lib');
+const { _ } = require('lodash');
+const { parametersValidation } = require('../../../helpers/validations.helper');
+const { lock, lockTypes } = require('../../../lock');
+const { cantSendTxErrorMessage } = require('../../../helpers/constants');
+const { mapTxReturn, getUtxosToFillTx } = require('../../../helpers/tx.helper');
+
+async function buildTxProposal(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  // build a transaction hex
+  const { wallet } = req;
+  const network = wallet.getNetworkObject();
+  const { outputs } = req.body;
+
+  const tokens = new Map();
```

Comment:
I feel most of this code is pretty much the same as the `send-tx` API, isn't it? Can we reuse it so we don't duplicate this code?

Reply:
I gave some though to this, but I still think we should keep the implementation separate.
The main reason would be for future implementations, I think we may add multisig and atomic-swap features to the `tx-proposal` subroute, so later the helper we will create to avoid duplicating code will only be used on the `send-tx` api, making the change seem not that useful.
What do you think?

Reply:
I'm not sure I agree. Even after we consolidate multisig and atomic swap APIs into this one, we will need all those parameters and this same logic for getting the correct utxos to create the proposal, won't we?

Reply:
It depends, the atomic-swap does not need to select utxos when creating the proposal.

Reply:
Does not need to but it might need. It's the same as the `send-tx` API. The user can manually select the input but we can have a utxo selection, then we would need the same logic in both to support querying by address and the common utxo selection.